mlflow/types/schema.py

```diff
@@ -116,84 +118,191 @@ def __repr__(self) -> str:
+        return cls(tensor_type, tensor_shape, kwargs["name"] if "name" in kwargs else None)
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, TensorSpec):
+            names_eq = (self.name is None and other.name is None) or self.name == other.name
+            return names_eq and self.type == other.type and self.shape == other.shape
+        return False
+
+    def __repr__(self) -> str:
+        if self.name is None:
+            return "({type}, {shape})".format(type=repr(self.type.name), shape=repr(self.shape))
+        else:
+            return "({name}, {type}, {shape})".format(
+                name=repr(self.name), type=repr(self.type.name), shape=repr(self.shape)
+            )
+
+
 class Schema(object):
     """
-    Specification of types and column names in a dataset.
+    Specification of a dataset.
 
-    Schema is represented as a list of :py:class:`ColSpec`. The columns in a schema can be named,
-    with unique non empty name for every column, or unnamed with implicit integer index defined by
-    their list indices. Combination of named and unnamed columns is not allowed.
+    Schema is represented as a list of :py:class:`ColSpec` or :py:class:`TensorSpec`. A combination
+    of `ColSpec` and `TensorSpec` is not allowed.
+
+    The dataset represented by a schema can be named, with unique non empty names for every input.
+    In the case of :py:class:`ColSpec`, the dataset columns can be unnamed with implicit integer
+    index defined by their list indices.
+    Combination of named and unnamed data inputs are not allowed.
     """
 
-    def __init__(self, cols: List[ColSpec]):
+    SchemaInput = Union[ColSpec, TensorSpec]
+
+    def __init__(self, dataset: List[SchemaInput]):
         if not (
-            all(map(lambda x: x.name is None, cols)) or all(map(lambda x: x.name is not None, cols))
+            all(map(lambda x: x.name is None, dataset))
+            or all(map(lambda x: x.name is not None, dataset))
         ):
             raise MlflowException(
                 "Creating Schema with a combination of named and unnamed columns "
-                "is not allowed. Got column names {}".format([x.name for x in cols])
+                "is not allowed. Got column names {}".format([x.name for x in dataset])
             )
-        self._cols = cols
+        if not (
+            all(map(lambda x: isinstance(x, TensorSpec), dataset))
+            or all(map(lambda x: isinstance(x, ColSpec), dataset))
+        ):
+            raise MlflowException(
+                "Creating Schema with a combination of {0} and {1} is not supported. "
+                "Please choose one of {0} or {1}".format(ColSpec.__class__, TensorSpec.__class__)
+            )
+        self._data_rep = dataset
 
     @property
-    def columns(self) -> List[ColSpec]:
-        """The list of columns that defines this schema."""
-        return self._cols
+    def data_rep(self) -> List[ColSpec]:
```

Comment:
the return type should be a List[Union[ColSpec,TensorSpec]] right? Or does TensorSpec inherit from ColSpec?